### PR TITLE
Location

### PR DESCRIPTION
**Pull request by jburel**
Handle location url with https
Tests adjusted, thanks @bramalingam 

Prior to the change, the ```list``` method entered in an infinite loop
All tests should now pass

### CODE REVIEW

File: components/formats-common/src/loci/common/Location.java [hunk 1]
```diff
@@ -363,18 +366,22 @@ public static void checkValidId(String id) throws IOException {
...
     }
-    ArrayList<String> files = new ArrayList<String>();
+    final List<String> files = new ArrayList<String>();
     if (isURL) {
       try {
         URLConnection c = url.openConnection();
         InputStream is = c.getInputStream();
         boolean foundEnd = false;
-
+        BufferedReader br = new BufferedReader(new InputStreamReader(is));
```
Comment (sbesson) [inline components/formats-common/src/loci/common/Location.java]: See https://ci.openmicroscopy.org/job/BIOFORMATS-5.1-merge-build/995/findbugsResult/new/

Comment (jburel) [inline components/formats-common/src/loci/common/Location.java]: useful comment!!!

**Comment by jburel** [comment]
removing exclude

**Comment by sbesson** [comment]
@jburel is someone giving a final test on this as it wasn't relisted?

**Comment by jburel** [comment]
Just need the tests to be green.

File: components/formats-common/src/loci/common/Location.java [hunk 2]
```diff
@@ -363,18 +366,23 @@ public static void checkValidId(String id) throws IOException {
...
+          buffer.append(input);
+        }
+        br.close();
+        String s = buffer.toString();
         while (!foundEnd) {
-          byte[] b = new byte[is.available()];
-          is.read(b);
-          String s = new String(b, Constants.ENCODING);
-          if (s.toLowerCase().indexOf("</html>") != -1) foundEnd = true;
+         if (s.toLowerCase().indexOf("</html>") != -1) foundEnd = true;
```
Comment (sbesson) [inline components/formats-common/src/loci/common/Location.java]: Small indentation problem - can be fixed later.

**Comment by sbesson** [comment]
Tests passing in Maven (Travis) and Ant (CI). Merging for 5.1.4.

**Pull request merged**